app/lucy/src/app/services/router.service.ts

```diff
@@ -168,4 +177,26 @@ export class RouterService {
     }
     return undefined;
   }
+
+  private preventReload() {
+    console.log('initialized prevent reload');
+    window.addEventListener(`beforeunload`, (event) => {
+      console.log('triggered');
+      if (this.isCreateRoute || this.isEditRoute) {
+        // Cancel the event as stated by the standard.
+        event.preventDefault();
+        // Chrome requires returnValue to be set.
+        event.returnValue = 'Your changes will be lost';
+        return 'Your changes will be lost';
+      }
+    });
+  }
+
+  private async shouldLeaveDialog(): Promise<boolean> {
+    if (this.isCreateRoute || this.isEditRoute) {
+      return this.alert.showConfirmation(`Are you sure?`, 'If you leave this page, your changes will be lost', `Leave Page`, `Stay`);
+    } else {
+      return true;
+    }
+  }
```

Comment:
rewrite like that 
private async shouldLeaveDialog(): Promise<boolean> { return (this.isCreateRoute || this.isEditRoute) ? this.alert.showConfirmation(`Are you sure?`, 'If you leave this page, your changes will be lost', `Leave Page`, `Stay`) : true}